### PR TITLE
ClusterControl.joined should cause joining; fix waiting issues

### PR DESCRIPTION
**Pull request by ktoso**
bug: We wrongly waited on the `self.node` to become up but should wait for the other node in `joined(node:)`

improvement: We should initiate joining when not already trying to and someone asks `try await joined(node:)` since the equivalent `join()` makes it seem like joined would do this as well

### CODE REVIEW

File: Sources/DistributedCluster/Cluster/ClusterControl.swift [hunk 1]
```diff
@@ -175,19 +192,33 @@ public struct ClusterControl {
...
     ///
     /// - Parameters
-    ///   - node: The node to be joined by this system.
+    ///   - endpoint: The endpoint to be joined by this system.
     ///   - within: Duration to wait for.
     ///
     /// - Returns `Cluster.Member` for the joined node.
     @discardableResult
     public func joined(endpoint: Cluster.Endpoint, within: Duration) async throws -> Cluster.Member? {
-        try await self.waitFor(self.node, .up, within: within)
```
Comment (ktoso) [inline Sources/DistributedCluster/Cluster/ClusterControl.swift:190]: actual bug here, we must wait for `endpoint` not `self.node` ;)

File: Sources/DistributedCluster/Cluster/ClusterControl.swift [hunk 2]
```diff
@@ -314,7 +347,7 @@ public struct ClusterControl {
                 throw Cluster.MembershipError(.notFound(node, in: membership))
             }
 
-            if atLeastStatus <= foundMember.status {
```
Comment (ktoso) [inline Sources/DistributedCluster/Cluster/ClusterControl.swift:317]: another mini bug; we should error out if the expected status is lower than the current one.

if they are the same, don't error -- we succeeded; the test covers this now

**Comment by ktoso** [comment]
The failure is `ClusterSingletonPluginClusteredTests.test_singletonByClusterLeadership_withLeaderChange` and I've actually been looking into it now, may solve. It's a bit tricky coming back to this after a while, but I'll figure it out -- it's an unexpected dead letter :)

**Comment by ktoso** [comment]
Failed: https://github.com/apple/swift-distributed-actors/issues/1124 needs investigation

**Comment by ktoso** [comment]
This improves things but we definitely still have some issues to look into here, I'll schedule time to work on those consistently now.

**Review by yim-lee — APPROVED** [review]

**Pull request merged**